datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -389,6 +384,7 @@ def create_source(
     node = Node(
         name=data.name,
         namespace=data.namespace,
+        display_name=data.display_name or f"{data.catalog}.{data.schema_}.{data.table}",
```

Comment:
I see the reason for doing this but I'm worried that it might be inconsistent if the user can set a custom display name. If we want to display `f"{data.catalog}.{data.schema_}.{data.table}"` for source nodes in the UI (which I think is a good idea), should we just build that concatenation client side?

Reply:
Hmm, I feel like it might even more confusing if we show a display name in the UI that doesn't match what's stored in the backend? Also this logic is actually just meant to replicate the existing functionality, where source nodes are automatically created with a display name of <catalog>.<schema>.<table> unless the user overrides it with a different display name. We can change this, but I think that's better off in a separate PR. And maybe we should make it so that users can never override the display name of a source node.

Reply:
> But maybe we should make it so that users can never override the display name of a source node, since I'm not sure it makes sense.

This summarizes my concern well. The purpose of display name feels a bit ambiguous to me. If users can never set it and it's derived from other attributes, then it seems it should be handled in the UI code. If it's something the user is meant to set, then I don't think we should systematically set it at all.

I'm leaning towards source nodes not having a display name at all (or alternatively, it's always ignored), then in the UI wherever we are showing a display name we do a check of the node type. If it's a source node, we define it at runtime as `${catalog}.${schema}.${table}`